### ISSUE TITLE
Setting Lua Version Fails (Windows 64x)

### ISSUE THREAD

**Issue by dcap0** [luarocks/luarocks#1459]
`luarocks config lua_version 5.3.6`

```
Error: LuaRocks 3.9.1 bug (please report at https://github.com/luarocks/luarocks/issues).    
Arch.: win32-x86_64 (fs_use_modules = true) (binary)
[string "src/luarocks/deps.lua"]:656: bad argument #1 to 'floor' (number expected, got nil)  
stack traceback:
        [C]: in function 'math.floor'
        [string "src/luarocks/deps.lua"]:656: in upvalue 'lua_h_exists'
        [string "src/luarocks/deps.lua"]:689: in upvalue 'find_lua_incdir'
        [string "src/luarocks/deps.lua"]:711: in function 'luarocks.deps.check_lua_incdir'   
        [string "src/luarocks/cmd/config.lua"]:243: in function 'luarocks.cmd.config.command'
        (...tail calls...)
        [C]: in function 'xpcall'
        [string "src/luarocks/cmd.lua"]:672: in function 'luarocks.cmd.run_command'
        [string "luarocks"]:34: in main chunk
        [C]: in ?
```

**Comment by forkingpath** [comment]
This also occurs with 5.4.4; it results in LUA_LIB showing as (not_found) because the internal variable gets rounded down to 5.4. I had to rename my library files from 'lua5.4.4.dll' to 'lua5.4.dll' in order for luarocks to build & install packages properly.

**Comment by hishamhm** [comment]
Fixed in aff9d641c60881d9fbb2e86bb5426a38fa4f1a59
